### ISSUE TITLE
robinhood v3: rbh-undelete query problem

### ISSUE THREAD

**Issue by geraldhofer** [cea-hpc/robinhood#68]
```
[root@tclin01 siteAB]# rbh-undelete -L /gdatat1/siteAB/c++
Using config file '/etc/robinhood.d/gdatat1.conf'.
2017/02/24 18:14:45 [6738/1] CheckFS | '/gdatat1' matches mount point '/gdatat1', type=lustre, fs=192.168.1.11@o2ib:/gdatat1
             rm_time,                        id,     type,       user,      group,       size,             last_mod,     lhsm.status,                                     path
2017/02/24 18:14:45 [6738/1] ListMgr | Unhandled error 1139: default conversion to DB_REQUEST_FAILED
2017/02/24 18:14:45 [6738/1] ListMgr | Error 7 executing query 'SELECT id,fullpath,uid,gid,size,blocks,last_access,last_mod,last_mdchange,type,mode,nlink,rm_time,lhsm_status,lhsm_archid,lhsm_uuid FROM SOFT_RM WHERE fullpath RLIKE BINARY '/gdatat1/siteAB/c++($|/.*)' ORDER BY rm_time ASC': Got error 'repetition-operator operand invalid' from regexp
2017/02/24 18:14:45 [6738/1] ListMgr | DB query failed in ListMgr_RmList line 907: query="SELECT id,fullpath,uid,gid,size,blocks,last_access,last_mod,last_mdchange,type,mode,nlink,rm_time,lhsm_status,lhsm_archid,lhsm_uuid FROM SOFT_RM WHERE fullpath RLIKE BINARY '/gdatat1/siteAB/c++($|/.*)' ORDER BY rm_time ASC",code=7, Got
2017/02/24 18:14:45 [6738/1] Undelete | ERROR: Could not retrieve removed entries from database.

[root@tclin01 siteAB]# rbh-undelete -L /gdatat1/siteAB/c\*
Using config file '/etc/robinhood.d/gdatat1.conf'.
2017/02/24 18:15:04 [6876/1] CheckFS | '/gdatat1' matches mount point '/gdatat1', type=lustre, fs=192.168.1.11@o2ib:/gdatat1
             rm_time,                        id,     type,       user,      group,       size,             last_mod,     lhsm.status,                                     path
 2017/02/24 18:10:59,    [0x2000013a0:0x62:0x0],     file,       root,       root,  750.59 KB,  2017/02/17 07:36:06,         synchro,                      /gdatat1/siteAB/c++
```

**Comment by tl-cea** [comment]
All user string are escaped using mysql_real_escape_string().
However, it looks like this function does not handle characters that have a meaning for regular expression handled by RLIKE. So extra escaping should be done to build requests using "RLIKE" criterion.

**Comment by tl-cea** [comment]
Fixed by commit 8ef267440d6abbafdf4cfd03bf3bf8a685a93ac6
